Frontend/Pages/KeyStage4Performance.cshtml.cs

```diff
@@ -0,0 +1,82 @@
+using System.Threading.Tasks;
+using Data;
+using Data.Models.KeyStagePerformance;
+using Frontend.ExtensionMethods;
+using Frontend.Models.Forms;
+using Frontend.Services.Interfaces;
+using Frontend.Services.Responses;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.Mvc.RazorPages;
+
+namespace Frontend.Pages
+{
+    public class KeyStage4Performance : PageModel
+    {
+        private readonly IGetInformationForProject _getInformationForProject;
+        private readonly IProjects _projectRepository;
+        public string ProjectUrn { get; private set; }
+        public string OutgoingAcademyUrn { get; private set; }
+        public string OutgoingAcademyName { get; private set; }
+        public string LocalAuthorityName { get; private set; }
+        public AdditionalInformationViewModel AdditionalInformation { get; private set; }
+        public List<KeyStage4> KeyStage4Results { get; private set; }
+        
+        public KeyStage4Performance(IGetInformationForProject getInformationForProject, IProjects projectRepository)
+        {
+            _getInformationForProject = getInformationForProject;
+            _projectRepository = projectRepository;
+        }
+        
+        public async Task<IActionResult> OnGetAsync(string id, bool addOrEditAdditionalInformation = false)
+        {
+            var projectInformation = await _getInformationForProject.Execute(id);
+
+            if (!projectInformation.IsValid)
+            {
+                return this.View("ErrorPage", projectInformation.ResponseError.ErrorMessage);
+            }
+
+            BuildPageModel(projectInformation, addOrEditAdditionalInformation);
+
+            return Page();
+        }
+
+        public async Task<IActionResult> OnPostAsync(string id, string additionalInformation)
+        {
+            var project = await _projectRepository.GetByUrn(id);
+
+            if (!project.IsValid)
+            {
+                return this.View("ErrorPage", project.Error.ErrorMessage);
+            }
+            
+            project.Result.KeyStage2PerformanceAdditionalInformation = additionalInformation;
+            await _projectRepository.Update(project.Result);
+            
+            return new RedirectToPageResult(nameof(KeyStage4Performance), 
+                "OnGetAsync", 
+                new { id }, 
+                "additional-information-hint");
+        }
+
+        private void BuildPageModel(GetInformationForProjectResponse projectInformation, bool addOrEditAdditionalInformation)
+        {
+            ProjectUrn = projectInformation.Project.Urn;
+            OutgoingAcademyUrn = projectInformation.OutgoingAcademy.Urn;
+            LocalAuthorityName = projectInformation.OutgoingAcademy.LocalAuthorityName;
+            OutgoingAcademyName = projectInformation.OutgoingAcademy.Name;
+            KeyStage4Results = projectInformation.EducationPerformance.KeyStage4Performance.Take(3).OrderByDescending(a => a.Year)
```

Comment:
you could probably move the right hand side of this into a private method in the class cos it's a bit much to read in here - probably like `GetLatestThreeResults` or something to that effect